src/lib/swapper/swappers/ThorchainSwapper/utils/makeSwapMemo/assertIsValidMemo.ts

```diff
@@ -69,7 +69,14 @@ export const assertIsValidMemo = (memo: string, chainId: ChainId, affiliateBps:
   const [, pool, address, limitComponent, affiliate, memoAffiliateBps] =
     memo.split(MEMO_PART_DELIMITER)
 
-  const buyAssetChainId = thorChainAssetToChainId.get(pool.split(POOL_PART_DELIMITER)[0])
+  if (!pool) throw new Error(`pools is required in memo: ${memo}`)
```

Comment:
Non-breaking change - the line below would throw anyway if we were trying to call `.split()` on `undefined`
<img width="486" alt="image" src="https://github.com/shapeshift/web/assets/17035424/63e8e1be-e86e-4cf4-ba45-b37966605c7d">

Reply:
The root cause here is we're missing `noUncheckedIndexedAccess` in our `tsconfig.json`. Might be worth checking the work needed to enable this as it'd prevent most cases but likely has thousands of violations

Reply:
Ah, I see your original comment

> since we can't do the `noUncheckedIndexedAccess` dance without having 500+ errors, this at least brings strictness for the specific